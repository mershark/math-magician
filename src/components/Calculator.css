body {
  display: flex;
  justify-content: center;
  background-color: #f5f5f5;
  margin: 0;
  font-family: Arial, sans-serif;
}

.App {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.calculator-app {
  width: 40%;
}

.quote-app {
  width: 60%;
  padding: 20px;
  background-color: #fff;
  box-shadow: 1px 2px 10px rgba(0, 0, 0, 0.1);
  border-radius: 5px;
}

.calculator {
  width: 100%;
  max-width: 300px;
  margin: 0 auto;
  border: 1px solid #e0e0e0;
  text-align: center;
  background-color: #fff;
  box-shadow: 1px 2px 10px rgba(0, 0, 0, 0.1);
  border-radius: 5px;
}

.calculator-display {
  font-size: 24px;
  padding: 15px 10px;
  display: flex;
  justify-content: flex-end;
  background-color: #858693;
  color: white;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
}

.calculator-grid {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 10px;
  padding: 10px;
}

.calculator-button {
  width: 100%;
  padding: 15px;
  font-size: 18px;
  background-color: #e0e0e0;
  cursor: pointer;
  border: 1px solid #c5c5c5;
  border-left: 1px solid #e0e0e0;
  border-radius: 5px;
  transition: background-color 0.2s ease-in-out;
}

.calculator-button:hover {
  background-color: #d2d2d2;
}

.operator-button {
  background-color: #f39148;
  border-right: 0;
  border-top: 0;
}

.zero-button {
  grid-column: span 2;
}

.quote-text {
  font-size: 20px;
  line-height: 1.5;
  color: #333;
}

.quote-author {
  margin-top: 10px;
  font-size: 18px;
  font-style: italic;
  color: #777;
}

.quote-loading {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100px;
  font-size: 18px;
  color: #666;
}

@media (max-width: 768px) {
  .App {
    flex-direction: column-reverse;
    align-items: center;
    gap: 20px;
  }

  .calculator-app {
    width: 100%;
    margin-left: 0;
  }

  .quote-app {
    width: 80%;
    margin-left: 10%;
    margin-right: 10%;
  }
}
