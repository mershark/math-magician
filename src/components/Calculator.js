import React, { useState } from 'react';
import './Calculator.css';
import Button from './Buttons';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [displayValue, setDisplayValue] = useState('0');
  const [calculatorState, setCalculatorState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleButtonClick = (buttonName) => {
    const newState = calculate(calculatorState, buttonName);
    // Update displayValue immediately after setting the state
    const newDisplayValue = newState.next || newState.total || '0';
    setDisplayValue(newDisplayValue);
    // Update calculator state after updating displayValue
    setCalculatorState(newState);
  };
  return (
    <div className="calculator">
      <div className="calculator-display">{displayValue}</div>
      <div className="calculator-grid">
        {/* Attaching the event handler to each Button */}
        <Button onClick={() => handleButtonClick('AC')}>AC</Button>
        <Button onClick={() => handleButtonClick('+/-')}>+/-</Button>
        <Button onClick={() => handleButtonClick('%')}>%</Button>
        <Button onClick={() => handleButtonClick('รท')}>รท</Button>
        <Button onClick={() => handleButtonClick('7')}>7</Button>
        <Button onClick={() => handleButtonClick('8')}>8</Button>
        <Button onClick={() => handleButtonClick('9')}>9</Button>
        <Button onClick={() => handleButtonClick('x')}>x</Button>
        <Button onClick={() => handleButtonClick('4')}>4</Button>
        <Button onClick={() => handleButtonClick('5')}>5</Button>
        <Button onClick={() => handleButtonClick('6')}>6</Button>
        <Button onClick={() => handleButtonClick('-')}>-</Button>
        <Button onClick={() => handleButtonClick('1')}>1</Button>
        <Button onClick={() => handleButtonClick('2')}>2</Button>
        <Button onClick={() => handleButtonClick('3')}>3</Button>
        <Button onClick={() => handleButtonClick('+')}>+</Button>
        <Button onClick={() => handleButtonClick('0')}>0</Button>
        <Button onClick={() => handleButtonClick('.')}>.</Button>
        <Button onClick={() => handleButtonClick('=')}>=</Button>
      </div>
    </div>
  );
};

export default Calculator;
