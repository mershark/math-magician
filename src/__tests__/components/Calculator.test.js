import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../../components/Calculator';

test('Calculator component renders correctly', () => {
  const { container } = render(<Calculator />);
  expect(container).toMatchSnapshot();
});

test('Calculator component handles button clicks correctly', () => {
  const { getByText, getByTestId } = render(<Calculator />);

  // Simulate button clicks
  fireEvent.click(getByText('1'));
  fireEvent.click(getByText('+'));
  fireEvent.click(getByText('2'));
  fireEvent.click(getByText('='));

  // Check if the displayValue is updated
  const displayValue = getByTestId('calculator-display');
  expect(displayValue.textContent).toBe('3');
});
