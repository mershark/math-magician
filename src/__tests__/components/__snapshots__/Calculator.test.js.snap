// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<div>
  <div
    class="calculator"
  >
    <div
      class="calculator-display"
    >
      0
    </div>
    <div
      class="calculator-grid"
    >
      <button
        class="calculator-button "
        type="button"
      >
        AC
      </button>
      <button
        class="calculator-button "
        type="button"
      >
        +/-
      </button>
      <button
        class="calculator-button "
        type="button"
      >
        %
      </button>
      <button
        class="calculator-button operator-button"
        type="button"
      >
        รท
      </button>
      <button
        class="calculator-button "
        type="button"
      >
        7
      </button>
      <button
        class="calculator-button "
        type="button"
      >
        8
      </button>
      <button
        class="calculator-button "
        type="button"
      >
        9
      </button>
      <button
        class="calculator-button operator-button"
        type="button"
      >
        x
      </button>
      <button
        class="calculator-button "
        type="button"
      >
        4
      </button>
      <button
        class="calculator-button "
        type="button"
      >
        5
      </button>
      <button
        class="calculator-button "
        type="button"
      >
        6
      </button>
      <button
        class="calculator-button operator-button"
        type="button"
      >
        -
      </button>
      <button
        class="calculator-button "
        type="button"
      >
        1
      </button>
      <button
        class="calculator-button "
        type="button"
      >
        2
      </button>
      <button
        class="calculator-button "
        type="button"
      >
        3
      </button>
      <button
        class="calculator-button operator-button"
        type="button"
      >
        +
      </button>
      <button
        class="calculator-button zero-button"
        type="button"
      >
        0
      </button>
      <button
        class="calculator-button "
        type="button"
      >
        .
      </button>
      <button
        class="calculator-button operator-button"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
